draft-irtf-cfrg-hpke.md

```diff
@@ -347,21 +347,27 @@ A set of algorithm identifiers for concrete instantiations of these
 primitives is provided in {{ciphersuites}}.  Algorithm identifier
 values are two bytes long.
 
-The following two functions are defined for a KDF to facilitate domain
-separation of calls as well as context binding:
+The following two functions are defined to facilitate domain separation of
+KDF calls as well as context binding:
 
 ~~~
 def LabeledExtract(salt, label, IKM):
-  labeledIKM = concat("RFCXXXX ", label, IKM)
+  labeledIKM = concat("RFCXXXX ", identifier, label, IKM)
   return Extract(salt, labeledIKM)
 
 def LabeledExpand(PRK, label, info, L):
-  labeledInfo = concat(I2OSP(L, 2), "RFCXXXX ", label, info)
+  labeledInfo = concat(I2OSP(L, 2), "RFCXXXX ", identifier, label, info)
   return Expand(PRK, labeledInfo, L)
 ~~~
 
 \[\[RFC editor: please change "RFCXXXX" to the correct number before publication.]]
 
+The value of `identifier` depends on where the KDF is used; it is assumed
+implicit from the implementation and not passed as parameter. If used
+inside a KEM algorithm, `identifier` MUST identify this KEM algorithm;
+if used in the remainder of HPKE, it MUST identify the entire ciphersuite
+in use. See sections {{dhkem}} and {{encryption-context}} for details.
```

Comment:
This seems likely to be confused.  Would it be clearer to add the identifier as a parameter to these methods, and explicitly pass in the right one?

Reply:
Note that the suite ID is a different length than the individual KEM ID.  Do we need to worry about collision with the label?

Reply:
> Note that the suite ID is a different length than the individual KEM ID. Do we need to worry about collision with the label?

Uh, good one. Yes, theoretically `aead_id` could collide with the labels `eae_prk` and `zz`. Not for the AEADs in the current document though: `hex(ord("z"))` is `0x7a` and `hex(ord("e"))` is `0x65`, and all current AEADs have 0s in the first byte.

We could have the `suite_id` for the KEMs be `concat(I2OSP(kem_id, 2), zero(4))` to avoid that forever.